pageserver: remove L0 flush upload wait

erikgrinaker opened the pull request:
## Problem

Previously, L0 flushes would wait for uploads, as a simple form of backpressure. However, this prevented flush pipelining and upload parallelism. It has since been disabled by default and replaced by L0 compaction backpressure.

Touches https://github.com/neondatabase/cloud/issues/24664.

## Summary of changes

This patch removes L0 flush upload waits, along with the `l0_flush_wait_upload`. This can't be merged until the setting has been removed across the fleet.

skyzh reviewed (approved):
A few tests need to be fixed, otherwise LGTM.

erikgrinaker commented:
Now that we've disabled this by default, I've updated this to simply remove the option entirely, We'll have to hold this until the new default has been deployed next week and we've removed the setting from the configs.
